src/RA_Dlg_GameLibrary.cpp

```diff
@@ -21,7 +21,13 @@
 #include "RA_httpthread.h"
 #include "RA_md5factory.h"
 
-#define KEYDOWN(vkCode) ((GetAsyncKeyState(vkCode) & 0x8000) ? true : false)
+
+inline constexpr auto CALLBACK KEYDOWN(int vk)->decltype((GetAsyncKeyState(vk) & 0x8000) ? true : false)
+{
+    return { (GetAsyncKeyState(vk) & 0x8000) ? true
+                                             : false };
+}
```

Comment:
I admit that I don't understand the `->decltype()` syntax, but it seems there should be a way to do this without duplicating the body of the function in the `decltype`. Can't you just use `decltype(bool)` or change the `auto` to an `bool`?

Reply:
> Do you see a disambiguation is important?
Yes, but in this case, why use `auto` and `decltype` when you can just explicitly say `bool`? My concern here is the duplication of code. This snippet can never return anything other than a bool, so why have the compiler deduce that?

Reply:
Sure, it's not a big deal.